packages/shared/src/hooks/useCompanionTrigger.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useCallback, useContext } from 'react';
+import { useFeatureIsOn } from '@growthbook/growthbook-react';
+import { FeedPostClick } from './feed/useFeedOnPostClick';
+import { Post } from '../graphql/posts';
+import { Features } from '../lib/featureManagement';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+
+// import { useExtensionPermission } from '../../../extension/src/companion/useExtensionPermission';
+
+type CompanionTriggerProps = {
+  post: Post;
+  index: number;
+  row: number;
+  column: number;
+};
+
+export type CompanionTrigger = {
+  postUrl?: string;
+  onFeedArticleClick: FeedPostClick;
+  onPostArticleClick: (e: React.MouseEvent) => Promise<void>;
+  activateCompanion: () => void;
+  openArticle: () => Promise<void>;
+  isOpen: boolean;
+  toggleOpen: (value: boolean) => void;
+};
+
+export default function useCompanionTrigger(
+  customPostClickHandler: (
+    post: Post,
+    index?: number,
+    row?: number,
+    column?: number,
+  ) => Promise<void>,
+): CompanionTrigger {
+  const isExtension = process.env.TARGET_BROWSER;
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [data, setData] = useState<CompanionTriggerProps>();
+  const [isOpen, toggleOpen] = useState(false);
+  const featureEnabled = useFeatureIsOn(
+    Features.EngagementLoopJuly2023Companion.toString(), // this feels very smelly, but otherwise complains because of the Features<string> type
+  );
+  // const { requestContentScripts, contentScriptGranted, isFetched } =
+  //   useExtensionPermission({
+  //     origin: 'companion modal permission button',
+  //     onPermission: () => {
+  //       if (data) {
+  //         readArticleHandler(data.post, data.index, data.row, data.column);
+  //       }
+  //     },
+  //   });
+
+  const handleToggle = useCallback((opened: boolean) => {
+    toggleOpen(opened);
+    const state = opened ? 'open' : 'close';
+
+    trackEvent({
+      event_name: `${state} companion permission popup`,
+      extra: JSON.stringify({ origin: 'automatic' }),
+    });
+
+    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const articleClickHandler = useCallback(
+    async (
+      e: React.MouseEvent,
+      { post, index, row, column }: Partial<CompanionTriggerProps> = {},
+    ) => {
+      if (!isExtension) {
+        // TODO: add contentScriptGranted || isFetched from useExtensionPermission
```

Comment:
Todo still needed?

Reply:
Yep, this and the 3 above you mentioned need to be enabled once we figure out how to go about using the `useExtensionPermission` hook -- see second note here: https://github.com/dailydotdev/apps/pull/2039#issuecomment-1676987569